packages/node/src/indexer/dynamic-ds.service.ts

```diff
@@ -94,17 +95,16 @@ export class DynamicDsService {
 
     assert(this.metaDataRepo, `Model _metadata does not exist`);
     await this.metaDataRepo.upsert(
-      { key: METADATA_KEY, value: JSON.stringify([...existing, dsParams]) },
-      { transaction: tx },
+      { key: METADATA_KEY, value: JSON.stringify([...existing, dsParams]) }
```

Comment:
Removing the tx here will lead to other problems. If the block fails to index for any reason then without a tx here the template will still be created and if indexing is resumed it will result in the template being created multiple times. 

I think a better option is to maintain `existing` in memory so we don't overwrite any other templates created in the same block

Reply:
@stwiname  
Thanks for the comment.
In our testing, as long as `{ transaction: tx }` is there, templates will still override each other.

Could you please suggest how we could prevent such overriding ?